terraform/cloud-platform-components/kiam.tf

```diff
@@ -0,0 +1,190 @@
+resource "tls_private_key" "ca" {
+  algorithm   = "ECDSA"
+  ecdsa_curve = "P384"
+}
+
+resource "tls_self_signed_cert" "ca" {
+  key_algorithm     = "${tls_private_key.ca.algorithm}"
+  private_key_pem   = "${tls_private_key.ca.private_key_pem}"
+  is_ca_certificate = true
+
+  validity_period_hours = 87600 // 10 years
+  early_renewal_hours   = 720   // 1 month
+
+  subject {
+    common_name = "Kiam CA"
+  }
+
+  allowed_uses = [
+    "cert_signing",
+    "crl_signing",
+  ]
+}
+
+resource "tls_private_key" "agent" {
+  algorithm   = "ECDSA"
+  ecdsa_curve = "P384"
+}
+
+resource "tls_cert_request" "agent" {
+  key_algorithm   = "${tls_private_key.agent.algorithm}"
+  private_key_pem = "${tls_private_key.agent.private_key_pem}"
+
+  subject {
+    common_name = "Kiam Agent"
+  }
+}
+
+resource "tls_locally_signed_cert" "agent" {
+  cert_request_pem   = "${tls_cert_request.agent.cert_request_pem }"
+  ca_key_algorithm   = "${tls_private_key.ca.algorithm}"
+  ca_private_key_pem = "${tls_private_key.ca.private_key_pem}"
+  ca_cert_pem        = "${tls_self_signed_cert.ca.cert_pem}"
+
+  validity_period_hours = 8760 // 1 year
```

Comment:
We need a way to manage the expiry of these certificates. 
Will kiam just stop working? 
Is it worth pattern matching a triggered alert?
When it expires, is it a case of re-running this Terraform file?

Reply:
As discussed:
- `kiam` will just stop working, indeed but with it, any `Pods` using IAM roles.
- We could use `blackbox_exporter` to get cert expiry in a prom metric and hook an alert on it. However, we will most likely drive this off https://github.com/ministryofjustice/cloud-platform/issues/587 so it's all done (hopefully) transparently. Should we just create a calendar event for now and raise an issue to address?
- Yes, re-running tf will renew & fix it.

Reply:
Yes, although not ideal the calendar entry will suffice.